src/System.Net.Http/tests/FunctionalTests/HttpProtocolTests.cs

```diff
@@ -413,37 +413,56 @@ public class HttpProtocolTests : HttpClientTestBase
             });
         }
 
+        public static IEnumerable<string> GetInvalidStatusLine()
+        {
+            yield return "HTTP/1.1 2345";
+            yield return "HTTP/A.1 200 OK";
+            yield return "HTTP/X.Y.Z 200 OK";
+            yield return "HTTP/1.A 200 OK";
+            yield return "HTTP/1.1 ";
+            yield return "HTTP/1.1 !11";
+            yield return "HTTP/1.1 a11";
+            yield return "HTTP/1.1 abc";
+            yield return "HTTP/1.1\t\t";
+            yield return "HTTP/1.1\t";
+            yield return "HTTP/1.1  ";
+            
+            // Only pass on .NET Core Windows & ManagedHandler.
```

Comment:
The `UseManagedHandler` detection cannot be applied in static field. Since for .NET Core Windows run, t[he ManagedHandler will be tested as well](https://github.com/dotnet/corefx/blob/95bfb895707d5756d4b9693286672780f0a6ba02/src/System.Net.Http/tests/FunctionalTests/ManagedHandlerTest.cs#L16), I didn't specify it.